tests/Feature/API/ApiAbilityEnforcementTest.php

```
@@ -0,0 +1,185 @@
+        Sanctum::actingAs($user, $otherAbilities === [] ? ['read'] : $otherAbilities);
+
+        $response = $this->json($method, route($routeName, $params));
+
+        $response->assertStatus(Response::HTTP_FORBIDDEN);
+    }
+
+    #[DataProvider('routeAbilityProvider')]
+    public function test_bearer_token_with_required_ability_is_not_denied_by_ability_check(
+        string $routeName,
+        string $method,
+        string $requiredAbility,
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$requiredAbility'. (undefined)

(UnusedFormalParameter)